src/verbs/device_context.rs

```diff
@@ -142,4 +150,89 @@ impl DeviceContext {
             }
         }
     }
+
+    pub fn legacy_query_gid_table(&self) -> Result<Vec<GidEntry>, String> {
+        let mut res = Vec::new();
+        let dev_attr = self.query_device().unwrap();
+        let mut gid_type;
+
+        for port_num in 1..(dev_attr.phys_port_cnt() + 1) {
+            let port_attr = self.query_port(port_num).unwrap();
+
+            if let Some(name) = self.name() {
+                for gid_index in 0..port_attr.gid_tbl_len() {
+                    let gid = self.query_gid(port_num, gid_index).unwrap();
+                    let netdev_index;
+
+                    if gid.is_zero() {
```

Comment:
`fe80:0000:0000:0000:0000:0000:0000:0000` should also be filtered out according to the https://github.com/Mellanox/mlnx-tools/blob/mlnx_ofed/sbin/show_gids

Reply:
https://github.com/linux-rdma/rdma-core/blob/9a3f3a5194b179567f97979323e6b3b9a90e2de5/libibverbs/cmd_device.c#L415
https://github.com/linux-rdma/rdma-core/blob/9a3f3a5194b179567f97979323e6b3b9a90e2de5/libibverbs/cmd_device.c#L214

As the comments say, I'm just mimicking the original C implementation, which only filtered out full zero gids.